accessor_image_format/accessor_image_format.md

```diff
@@ -0,0 +1,169 @@
+32-bit unsigned integers, or 16-bit floating points, which are in turn converted
+to the types of the channels in the image. Because of this, backends that cannot
+query the types of the channels in the image cannot determine the necessary
+conversions to do.
+
+## Overview
+
+This extension adds `sycl::short4`, `sycl::ushort4`, `sycl::schar4`, and
+`sycl::uchar4` to the list of types supported for `dataT` for image accessors.
+In order to carry information about the format of the image, the
+`image_format` enumerator from SYCL 2020 is adopted, given as
+
+```c++
+namespace sycl {
+enum class image_format : unsigned int {
+    r8g8b8a8_unorm,
+    r16g16b16a16_unorm,
+    r8g8b8a8_sint,
+    r16g16b16a16_sint,
+    r32b32g32a32_sint,
+    r8g8b8a8_uint,
+    r16g16b16a16_uint,
+    r32b32g32a32_uint,
+    r16b16g16a16_sfloat,
+    r32g32b32a32_sfloat,
+    b8g8r8a8_unorm,
+};
+} // namespace sycl
+```
+
+and mappings from `image_format` to the corresponding accessor data type is
+defined as
+
+```c++
+
+namespace sycl {
+template<image_format> struct image_format_type;
+
+template<> struct image_format_type<image_format::r8g8b8a8_unorm> {
+    using type = float4;
+};
+template<> struct image_format_type<image_format::r16g16b16a16_unorm> {
+    using type = float4;
+};
+template<> struct image_format_type<image_format::r8g8b8a8_sint> {
+    using type = schar4;
+};
+template<> struct image_format_type<image_format::r16g16b16a16_sint> {
+    using type = short4;
+};
+template<> struct image_format_type<image_format::r32b32g32a32_sint> {
+    using type = int4;
+};
+template<> struct image_format_type<image_format::r8g8b8a8_uint> {
+    using type = uchar4;
+};
+template<> struct image_format_type<image_format::r16g16b16a16_uint> {
+    using type = ushort4;
+};
+template<> struct image_format_type<image_format::r32b32g32a32_uint> {
+    using type = uint4;
+};
+template<> struct image_format_type<image_format::r16b16g16a16_sfloat> {
+    using type = half4;
+};
+template<> struct image_format_type<image_format::r32g32b32a32_sfloat> {
+    using type = float4;
+};
+template<> struct image_format_type<image_format::b8g8r8a8_unorm> {
+    using type = float4;
+};
+
+template<image_format format>
+using image_format_t = typename image_format_type<format>::type;
+} // namespace sycl
+```
+
+With these, a new overload of `get_access` is added to `image` defined as
+
+```c++
+template <image_format format, access::mode accessMode>
+accessor<image_format_t<format>, dimensions, accessMode, access::target::image>
+get_access(handler & commandGroupHandler);
+```
+
+Calling `read` or `write` on an image accessor retrieved through the new
+`get_access` above causes undefined behaviour unless the `image_channel_type` of
+the underlying image and the `image_format` given to `get_access` is one of the
+combinations shown in the following table:
+
+|  `image_channel_type `  |     `image_format`    |
```

Comment:
It would be useful to specify the type of `image_format_type_t` for each value of `image_format` as well to save cross-referencing it.

Reply:
I don't think this would be the table for it, but I could add a new table with the mappings if that would be useful?